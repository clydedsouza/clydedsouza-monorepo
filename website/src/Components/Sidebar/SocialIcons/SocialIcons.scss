@import '../../../Sass/variables';

.social-icons {
  font-size: 18px;
  font-weight: 100;
  margin: 9px 0 18px 0;

  span {
    margin: 0 2px;

    a {
      &:hover,
      &:focus {
        border-bottom: none;
      }
    }

    i {
      font-size: 24px;
      margin: 0 0.5vw;
      color: white;

      @include for-tablets-down {
        margin: 0 3vw;
      }

      &:first-child {
        margin-left: 0;
      }
    }
  }
}
