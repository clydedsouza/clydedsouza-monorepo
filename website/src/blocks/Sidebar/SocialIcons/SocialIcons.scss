@import "../../../styles/variables";

.social-icons {
  font-size: 18px;
  font-weight: 100;
  margin: 9px 0 18px 0;

  span {
    margin: 0 2px;

    a {
      &:hover,
      &:focus {
        border-bottom: none;
      }
    }

    svg {
      width: 24px;
      margin: 0 0.8vw;
      fill: white;
      vertical-align: middle;
      transition: all 0.2s linear 0.2s;

      &:hover,
      &:focus {
        opacity: 0.95;
      }

      &.linkedin-svg {
        width: 22px;
      }

      &.envelope-svg {
        width: 26px;
      }

      &.twitter-svg,
      &.youtube-svg {
        width: 28px;
      }

      @include for-tablets-down {
        margin: 0 3vw;
      }

      &:first-child {
        margin-left: 0;
      }
    }
  }
}
