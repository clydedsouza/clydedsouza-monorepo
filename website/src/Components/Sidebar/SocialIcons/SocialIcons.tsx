import { sendLinkClickedEvent } from '../../../Api/Analytics'
import { AnalyticsLinkType } from '../../../Api/IAnalytics'
import Envelope from '../../Icons/Envelope'
import GitHub from '../../Icons/GitHub'
import LinkedIn from '../../Icons/LinkedIn'
import Medium from '../../Icons/Medium'
import Twitter from '../../Icons/Twitter'
import YouTube from '../../Icons/YouTube'
import './SocialIcons.scss'

const LINKEDIN_PROFILE = 'https://linkedin.com/in/clydedz/'
const TWITTER_PROFILE = 'https://twitter.com/clydedz'
const YOUTUBE_PROFILE = 'https://www.youtube.com/@clydedz'
const GITHUB_PROFILE = 'https://github.com/ClydeDz/'
const MEDIUM_PROFILE = 'https://medium.com/@clydedz'
const EMAIL_ADDRESS = 'mailto:clyde@clydedsouza.net'

function SocialIcons() {
  return (
    <div className="social-icons">
      <span>
        <a
          href={LINKEDIN_PROFILE}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: LINKEDIN_PROFILE,
              type: AnalyticsLinkType.SocialIcons,
            })
          }
        >
          <LinkedIn />
        </a>
      </span>
      <span>
        <a
          href={TWITTER_PROFILE}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: TWITTER_PROFILE,
              type: AnalyticsLinkType.SocialIcons,
            })
          }
        >
          <Twitter />
        </a>
      </span>
      <span>
        <a
          href={YOUTUBE_PROFILE}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: YOUTUBE_PROFILE,
              type: AnalyticsLinkType.SocialIcons,
            })
          }
        >
          <YouTube />
        </a>
      </span>
      <span>
        <a
          href={GITHUB_PROFILE}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: GITHUB_PROFILE,
              type: AnalyticsLinkType.SocialIcons,
            })
          }
        >
          <GitHub />
        </a>
      </span>
      <span>
        <a
          href={MEDIUM_PROFILE}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: MEDIUM_PROFILE,
              type: AnalyticsLinkType.SocialIcons,
            })
          }
        >
          <Medium />
        </a>
      </span>
      <span>
        <a
          href={`${EMAIL_ADDRESS}?Subject=Hi%20Clyde%21&Body=Hi%20Clyde,%20%0D%0DI%20wanted%20to%20get%20in%20touch%20with%20you.%0D%0DThanks,%20%0D%0DSent%20from%20https%3A%2F%2Fclydedsouza.net`}
          target="_blank"
          rel="noreferrer"
          onClick={() =>
            sendLinkClickedEvent({
              link: EMAIL_ADDRESS,
              type: AnalyticsLinkType.SocialIcons,
            })
          }
        >
          <Envelope />
        </a>
      </span>
    </div>
  )
}

export default SocialIcons
