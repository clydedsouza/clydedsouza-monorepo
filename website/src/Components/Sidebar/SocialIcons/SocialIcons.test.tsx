import { render } from '@testing-library/react'
import SocialIcons from './SocialIcons'

describe('Social icons', () => {
  it('should render social icons', () => {
    render(<SocialIcons />)
    expect(document.body).toMatchSnapshot()
  })
})
