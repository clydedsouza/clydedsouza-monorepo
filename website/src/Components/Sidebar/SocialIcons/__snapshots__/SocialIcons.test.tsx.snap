// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social icons should render social icons 1`] = `
<body>
  <div>
    <div
      class="social-icons"
    >
      <span>
        <a
          href="https://linkedin.com/in/clydedz/"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="linkedin-svg"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
            />
          </svg>
        </a>
      </span>
      <span>
        <a
          href="https://twitter.com/clydedz"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="twitter-svg"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
            />
          </svg>
        </a>
      </span>
      <span>
        <a
          href="https://www.youtube.com/@clydedz"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="youtube-svg"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
            />
          </svg>
        </a>
      </span>
      <span>
        <a
          href="https://github.com/ClydeDz/"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="github-svg"
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </a>
      </span>
      <span>
        <a
          href="https://medium.com/@clydedz"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="medium-svg"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M180.5,74.262C80.813,74.262,0,155.633,0,256S80.819,437.738,180.5,437.738,361,356.373,361,256,280.191,74.262,180.5,74.262Zm288.25,10.646c-49.845,0-90.245,76.619-90.245,171.095s40.406,171.1,90.251,171.1,90.251-76.619,90.251-171.1H559C559,161.5,518.6,84.908,468.752,84.908Zm139.506,17.821c-17.526,0-31.735,68.628-31.735,153.274s14.2,153.274,31.735,153.274S640,340.631,640,256C640,171.351,625.785,102.729,608.258,102.729Z"
            />
          </svg>
        </a>
      </span>
      <span>
        <a
          href="mailto:clyde@clydedsouza.net?Subject=Hi%20Clyde%21&Body=Hi%20Clyde,%20%0D%0DI%20wanted%20to%20get%20in%20touch%20with%20you.%0D%0DThanks,%20%0D%0DSent%20from%20https%3A%2F%2Fclydedsouza.net"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="envelope-svg"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
            />
          </svg>
        </a>
      </span>
    </div>
  </div>
</body>
`;
